mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -1990,8 +1996,21 @@ static LogicalResult vectorizeLinalgOpPrecondition(
   // TODO: isaConvolutionOpInterface that can also infer from generic
   // features. But we will still need stride/dilation attributes that will be
   // annoying to reverse-engineer...
-  if (isa<ConvolutionOpInterface>(linalgOp.getOperation()))
+  if (isa<ConvolutionOpInterface>(linalgOp.getOperation())) {
+    // Create a dummy rewriter first, a rewriter is not required for
+    // validation
+    IRRewriter dummyBuilder(linalgOp.getContext());
```

Comment:
It's a bit weird that we need an IRRewriter to check the vectorization preconditions of a convolution. We may want to decouple the preconditions from the vectorization itself to the extent possible.

This function is also growing too much so it'd be good to create a dedicated `vectorizeConvolutionPrecondition`.

Reply:
I've got rid of it in 74a8986. I don't plan to bring it back in a future commit too because it does look hacky. Marking this as resolved.